services/web/client/source/class/qxapp/auth/core/Utils.js

```diff
@@ -0,0 +1,58 @@
+/* global document */
+
+qx.Class.define("qxapp.auth.core.Utils", {
+  type: "static",
+
+  statics:
+  {
+    checkPasswordSecure:  function(password, itemForm) {
+      const isValid = password !== null && password.length > 2;
+      if (!isValid) {
+        const msg = qx.locale.Manager.tr("Please enter a password at with least 3 characters.");
+        itemForm.setInvalidMessage(msg);
+      }
+      return isValid;
+    },
+
+    checkSamePasswords: function(passwordField1, passwordField2) {
+      const isValid = passwordField1.getValue() == passwordField2.getValue();
+      if (!isValid) {
+        [passwordField1, passwordField2].forEach(pass => {
+          pass.set({
+            invalidMessage: qx.locale.Manager.tr("Passwords do not match"),
+            valid: false
+          });
+        });
+      }
+      return isValid;
+    },
+
+    findGetParameter: function(parameterName) {
+      let result = null;
+      const params = window.location.search.substr(1).split("&");
```

Comment:
this seems dangerous as a rule, reading data of the url in the browser.  if this is necessary at all, I would place parameters into the part following the # character as this is explicitly thought to be the part responsible for navigating inside the page using anchor tags.

Reply:
Now we use fragment as ``#page=reset-password;code=123456`` so browser is not reloaded and front-end switches to the right page

![image](https://user-images.githubusercontent.com/32402063/50225546-282a7580-03a1-11e9-8afe-3ff78e72f7ec.png)